tmc-langs-util/src/main/java/fi/helsinki/cs/tmc/langs/util/Main.java

```diff
@@ -0,0 +1,161 @@
+            + "\nOptions:\n"
+            + " --checkstyle <exercise path> <output path>\t\tRun checkstyle or similar plugin to project if applicable.\n"
+            + " --help\t\t\t\t\t\t\tDisplay help information.\n"
+            + " --preparesolution <exercise path>\t\t\tPrepare a presentable solution from the original.\n"
+            + " --preparestub <exercise path>\t\t\t\tPrepare a stub exercise from the original.\n"
+            + " --runtests <exercise path> <output path>\t\tRun the tests for the exercise.\n"
+            + " --scanexercise <exercise path> <output path>\t\tProduce an exercise description of an exercise directory.\n");
+        System.exit(0);
+    }
+
+    private static void run(String[] args) {
+        Map<String, Integer> commands = getCommands();
+        String command = args[0];
+        Integer pathsCount = commands.get(command);
+
+        if (pathsCount == null) {
+            printHelp();
+        } else if (!pathsCount.equals(args.length - 1)) {
+            System.out.println("ERROR: wrong argument count for " + command
+                + " expected " + pathsCount + " got " + (args.length - 1));
+            printHelp();
+        }
+
+        Map<String, Path> paths = parsePaths(args, pathsCount);
+
+        switch (command) {
+            case "--help":
+                printHelp();
+                break;
+            case "--checkstyle":
+                runCheckCodeStyle(paths);
+                break;
+            case "--scanexercise":
+                runScanExercise(paths);
+                break;
+            case "--runtests":
+                runTests(paths);
+                break;
+            case "--preparestub":
+                runPrepareStub(paths);
+                break;
+            case "--preparesolution":
+                runPrepareSolution(paths);
+                break;
+            default:
+                printHelp();
+                break;
+        }
+    }
+
+    private static Map<String, Integer> getCommands() {
+        //Command name and required argument count
+        Map<String, Integer> commands = new HashMap<>();
+        commands.put("--help", 0);
+        commands.put("--checkstyle", 2);
+        commands.put("--scanexercise", 2);
+        commands.put("--runtests", 2);
+        commands.put("--preparestub", 1);
+        commands.put("--preparesolution", 1);
+        return commands;
+    }
+
+    private static void runCheckCodeStyle(Map<String, Path> paths) {
+        Optional<ValidationResult> validationResult = executor.runCheckCodeStyle(paths.get(EXERCISE_PATH));
+
+        if (validationResult.isPresent()) {
+            JsonWriter.writeObjectIntoJsonFormat(validationResult.get(), paths.get(OUTPUT_PATH));
+            System.out.println("Codestyle report can be found at " + paths.get(OUTPUT_PATH));
+        } else {
+            System.out.println("ERROR: Something went wrong, could not output validation result!");
```

Comment:
Please print all status messages to `System.err`. It's generally good practice, and it'd be nice to try to keep `System.out` clean in case we ever want a command that outputs something like JSON to stdout.